cmd/libsnap-confine-private/facts.c

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "facts.h"
+
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "cleanup-funcs.h"
+#include "string-utils.h"
+#include "utils.h"
+
+char *sc_load_facts(const char *fname)
+{
+	FILE *f SC_CLEANUP(sc_cleanup_file) = NULL;
+	char *buf_copy;
+	size_t nread;
+	char buf[16 * 1024 + 1];
+
+	f = fopen(fname, "rt");
+	if (f == NULL) {
+		if (errno == ENOENT) {
+			return NULL;
+		}
+		die("cannot open facts file %s", fname);
+	}
+
+	nread = fread(buf, 1, sizeof buf, f);
+	if (!feof(f)) {
```

Comment:
I think we should check `ferror(f)` first?

Reply:
The intent was to check that we reached the end of the file (that the read got all of the data). If for whatever reason we did not it's an error and we bail out.